NGS2_WITNESS_Cycle1_experiment1_prep.R

```diff
@@ -0,0 +1,422 @@
+    if(rename) {
+        names(tmp)[which(names(tmp) %in% 'empanel_id')] <- rename_var
+    }
+    return(tmp)
+}
+
+read_input_files <- function(dir) {
+    # identifies all csv files in a directory, reads then into r, and appends them
+    res <- list()
+    files <- list.files(dir, pattern = '.csv$')
+    sources <- strsplit(files, split = '_')
+    for(i in 1:length(files)) {
+        tmp <- read.csv(paste(dir, files[i], sep = '/'), header = TRUE,
+                        sep = ',', stringsAsFactors = FALSE)
+        tmp$source <- paste(sources[[i]][1:2], collapse = '_')
+        res[[i]] <- tmp
+        names(res)[i] <- paste(sources[[i]][1:2], collapse = '_')
+    }
+    return(res)
+}
+
+# Load raw breadboard data
+exp1 <- read_input_files(paste0('NGS2-Cycle', CYCLE, '-Experiment1/data'))
+
+# gather empanelment/breadboard crosswalk information
+empanel_bb_xwalk <- build_empanelment_bb_xwalk('oms_url_upload_20170821_1145.txt')
+
+# EXPERIMENT 1
+# cooperation dataset
+# create subsets for manipulation
+conn <- create_subsets(exp1, 'Connected', 'player', strip_practice = FALSE)
+act <- create_subsets(exp1, 'cooperationEvent', 'pid|round',
+                      strip_practice = TRUE)
+chng <- create_subsets(exp1, 'rewiringEvent', 'pid|nid|round',
+                       strip_practice = TRUE)
+chng <- lapply(chng, function(x) {
+    if(nrow(x) > 0) {
+        x$action_num <- ifelse(x$action == 'maintainConnection', 0,
+                               ifelse(x$action == 'breakConnection', -1, 1))
+    }
+    return(x)
+})
+
+# start building the `cooperation` output file
+coop <- build_cooperation(act)
+
+# calculate actor moves by round
+round_changes <- changes_by_round(chng, conn)
+nbr_connections_round <- calculate_connections(round_changes)
+
+# finish off the `cooperation` dataset
+coop <- mapply(function(x, y, exp) {
+    if(nrow(y) > 0) {
+        return(merge(x, y, by = c('round', 'pid')))
+    } else {
+        conns <- dcast(subset(exp, event == 'Connected'), id ~ data.name,
+                       value.var = 'data.value')
+        x$num_neighbors <- NA
+        for(i in 1:max(x$round)) {
+            x <- get_neighbor_number(x, conns, i)
+        }
+        return(x[, c('round', 'pid', 'action', 'group_size',
+                     'previous_decision', 'num_neighbors')])
+    }
+}, coop, nbr_connections_round, exp1, SIMPLIFY = FALSE)
+
+# add in parameters from dataset for completeness
+coop <- do.call(rbind, mapply(function(x, name, exp) {
+    exp_cond <- as.numeric(exp$data.value[exp$event == 'initParameters' &
+                                          exp$data.name == 'k'])
+    x$session <- name
+    x$condition <- ifelse(exp_cond == 0, 'Static',
+                          ifelse(exp_cond == .1, 'Viscous',
+                          ifelse(exp_cond == .3, 'Fluid',
+                          ifelse(exp_cond == 1, 'Random', 'Other'))))
+    return(x)
+}, coop, as.list(names(coop)), exp1, SIMPLIFY = FALSE))
+
```

Comment:
- Please add "fluid_dummy" variable: 

exp1_cooperation$fluid_dummy[exp1_cooperation$condition=="Fluid"] <- 1
exp1_cooperation$fluid_dummy[exp1_cooperation$condition=="Random" |
                             exp1_cooperation$condition=="Viscous" |
                               exp1_cooperation$condition=="Static"] <- 0

Reply:
You got it.